arangod/RocksDBEngine/ReplicatedRocksDBTransactionState.cpp

```diff
@@ -100,37 +101,45 @@ futures::Future<Result> ReplicatedRocksDBTransactionState::doCommit() {
   }};
 
   // Due to distributeShardsLike, multiple collections can have the same log
-  // leader. We need to make sure that we only commit once per log.
-  std::unordered_set<replication2::LogId> logs;
-
+  // leader. In this case, we are going to commit the same transaction multiple
+  // times in the same log. This is ok, because followers know how to handle
+  // this situation.
```

Comment:
That would be pretty good, and would fix a big issue that we're currently faced with:
Say we have two shards (distribute-like). We replicate and commit successfully on one, but fail to replicated on the other. That's messed up, and we can't do much about it currently.
Having to deal with one replicated log instead of N collections would mean there's only one point of failure (at least replication-wise).